src/main/java/pl/cleankod/util/ExchangeRateCache.java

```diff
@@ -0,0 +1,30 @@
+package pl.cleankod.util;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ExchangeRateCache {
+
+    private final Map<String, CachedData> exchangeRateCache = new ConcurrentHashMap<>();
```

Comment:
This in-memory cache implementation does not have a maximum size limit. Can you stop any potential issues because of that?